scipy/optimize/optimize.py

```diff
@@ -1465,7 +1465,8 @@ def _minimize_newtoncg(fun, x0, args=(), jac=None, hess=None, hessp=None,
             A = fhess(*(xk,) + args)
             hcalls = hcalls + 1
 
-        while numpy.add.reduce(numpy.abs(ri)) > termcond:
+        k2 = 0
+        while (numpy.add.reduce(numpy.abs(ri)) > termcond) and (k2 < maxiter):
```

Comment:
`maxiter` is a not good limiting value here. CG iterations converge in at most `number of variables` iterations, let's put that value here.

Let's remove parentheses.